### DISCUSSION TITLE
would be possible to export the data in a JSON format

### DISCUSSION THREAD

**Discussion by FlaviuRadulescu** [Syllo/nvtop#219] [Ideas]
Would be great if we could push the data into grafana, so I was wondering how hard would be to run the nvtop with a parameter that would return the current state in a JSON format which could then be pushed into grafana/prometheus.

We could easily have like a cron or something to interogate nvtop and export the data out, or to just save it in a log file in a JSON format (running as a demon without a UI, or just run once and return the output)

**Comment by Self-Perfection** [comment]
I came here with the same plea.

In my case I would like to monitor VRAM usage distribution by different applications. At the moment I can monitor total VRAM usage by radeontop as it has machine readable output:
```
$ radeontop --limit 1 --dump -
Dumping to -, line limit 1.
1715980978.357812: bus 05, gpu 3.33%, ee 0.00%, vgt 0.83%, ta 2.50%, sx 3.33%, sh 0.83%, spi 3.33%, sc 2.50%, pa 0.83%, db 2.50%, cb 2.50%, vram 82.03% 1660.54mb, gtt 6.90% 476.26mb, mclk 74.42% 0.893ghz, sclk 14.29% 0.200ghz
```

But I would like to monitor per application.

Is there any other tool that can monitor VRAM usage per process?
